lib/server.js

```diff
@@ -0,0 +1,65 @@
+const app = require('express')();
+const redis = require('redis');
+const bodyParser = require('body-parser').json();
+const shortid = require('shortid');
+const url = require('url');
+
+const API_HOST = process.env.WHISPER_HOST || 'localhost';
+const API_PORT = process.env.WHISPER_PORT || 5000;
+
+// create a new redis client and connect to our local redis instance
+const REDIS_PORT = process.env.REDIS_PORT || 6379;
+const REDIS_HOST = process.env.REDIS_HOST || 'localhost';
+const client = redis.createClient(REDIS_PORT, REDIS_HOST);
+
+client.on('error', err => {
+  console.log("Error " + err);
+});
+client.on('error', error => console.error(`redis error: ${error}`));
+client.on('end', () => console.log('Redis: connection closed'));
+client.on('warning', msg => console.log(`Redis warning: ${msg}`));
+
+app.set('host', API_HOST);
+app.set('port', API_PORT);
+
+app.get('/whisper/:id', (req, res) => getWhisper(req, res, client));
+app.post('/whisper', bodyParser, (req, res) => postWhisper(req, res, client));
+
+app.listen(app.get('port'), () => {
+  console.log('Server listening on port: ', app.get('port'));
+});
+
+function getWhisper(req, res, client) {
+  const id = req.params.id;
+
+  client.get(id, (err, result) => {
```

Comment:
Should return 404 on result === null and maybe some generic 400 message if redis returns an error.